pure.zsh

```diff
@@ -324,6 +324,11 @@ prompt_pure_async_git_fetch() {
 	# Set SSH `BachMode` to disable all interactive SSH password prompting.
 	export GIT_SSH_COMMAND="${GIT_SSH_COMMAND:-"ssh"} -o BatchMode=yes"
 
+	# if gpg-agent is setup to handle ssh keys for git fetch; then
+	# make sure it uses this zpty instead of corrupting the parent tty
+	# solves issue #366
+	export GPG_TTY=$TTY
```

Comment:
This unfortunately doesn't solve the problem.

The tty in the context of this function's execution is different from the user's current tty, so the same problem occurs (pinentry on a different tty, mangled input).

There is a trivial way to verify it: add `touch /tmp/mytty && echo $TTY >> /tmp/mytty` before this line, and check the contents of `/tmp/mytty` after changing directory to a Git repository.
E.g. when my tty is `/dev/ttys000`, the tty seen by pure's async functions is `/dev/ttys001`.

Reply:
The tty in this context is meant to be different, because pure lauches this function `prompt_pure_async_git_fetch` asynchroneoulsy in a different tty than the main one.

If the issue still occurs then possibly you didn't setup the ssh config in `~/.ssh/config` to always update the gpg agent with the current tty like mentioned in the archlinux wiki [Configure pinentry to use the correct TTY
](https://wiki.archlinux.org/title/GnuPG#Configure_pinentry_to_use_the_correct_TTY).

Reply:
You're right, I didn't have `Match host * exec "gpg-connect-agent UPDATESTARTUPTTY /bye"` inside my SSH config. Thanks for the tip!

That seems to eliminate the issue indeed. I still see a pinentry process owned by gpg-agent, but it seems like your second commit is supposed to fix it. I'll try it separately and report.

Reply:
I would suggest mentioning this command or referencing the page you linked somewhere, because this config is external to Pure, yet critical to be able to use Pure with such SSH setup.

I can volunteer to update the wiki with a section dedicated to GPG-related issues such as this one.

Reply:
I guess by wiki you mean the [pure wiki](https://github.com/sindresorhus/pure/wiki/)? Should it be about how to setup gpg-agent as a replacement for ssh-agent or should it be just a note about the updating the tty in ssh config?

Reply:
Yes the Pure wiki. I would just add a note about the necessity to add `Match host * exec ...` to the SSH config, because this might be quite unconventional (unless you are used to having daemons using SSH in the background). At least for me, updating gpg-agent's TTY in `.zshrc` was always enough, until I ran into this issue with Pure.

Reply:
Can you please review this: https://github.com/sindresorhus/pure/wiki/Advise-on-using-%60gpg-agent%60-as-an-alternative-%60ssh-agent%60. Feel free to edit as you see fit.